### PR TITLE
PHP7.0 compatibility

### PR DESCRIPTION
**Pull request by samuel20miglia**
Hello guys,

i know your work is good, but in the real world we need to work with a multiple installation types.

Please Add php7.0 support.

Thank you so much

### CODE REVIEW

**Comment by afoucret** [comment]
Hi @samuel20miglia,

There are several reason for us to support only PHP >= 7.1.

The first one is code quality. Even if we can live with our constants exposed as public, the nullable type return hint is a lot more powerful and grant us a greater type safety.
In your example, you proposed to remove replace `?string` by `string` in the `LanguageResolver` implementation. It would be wrong, because we may eventually return `null` (universal language in App Search). The right answer to support PHP 7.0 would be to remove the return type hint. 

We decided to use return type hint heavily because it improves the code quality and makes tests easier to write.

Actually the return type hints are now used in Magento core code ([an example in their ElasticSearch module](https://github.com/magento/magento2/blob/2.3/app/code/Magento/Elasticsearch6/Model/Adapter/FieldMapper/Product/FieldProvider/FieldName/Resolver/DefaultResolver.php)). That is one of the main reason why Magento 2.3 does not support PHP 7.0 anymore. Our decision is consistent with the practice of Magento team.

Last but not least, PHP 7.0 as reached EOL it means that it will not receive any security update. I think it is an important point to consider for us.

We could remove all our nullable type hints but it would:
- require to rewrite a lot of code and tests.
- have an impact on the code quality

**TL;DR:**

Because last versions of Magento does not support PHP 7.0, I feel like if the gain is too not very important compared to the advantage of having PHP 7.1 features available.
Instead, I would advocate to modify composer.json to make it clear that we don't support PHP 7.0.

**Comment by samuel20miglia** [comment]
you write this one on your Readme

> Magento >= 2.2.x installed and running

**Comment by afoucret** [comment]
Hi @samuel20miglia,

You are right about the documentation. We need to improve it to make it clear that we support only PHP >= 7.1. PR #55 is intended to fix it.

That being said, I wanted to know how do you feel with the point above and if it is something that could prevent you to use the module?
